docs: [IAC-3374]: IaCM Module testing (#9820)
harness/developer-hub: docs/infra-as-code-management/iacm-features/module-registry/module-registry-testing.md

---
title: Module Registry Testing
description: Learn how to add module registry testing to your Harness IaCM Pull Requests
sidebar_position: 50
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> Module Registry Testing automates testing for infrastructure modules, ensuring reliability and functionality while avoiding workspace credit consumption.

Reviewer: ```suggestion
Module Registry Testing helps with the testing for your modules, ensuring reliability and functionality while avoiding workspace credit consumption.
```

Reviewer: Uri may not want to say anything about the credit consumption, check with him

## Common use case

> 1. When you [register a module](/docs/infra-as-code-management/iacm-features/module-registry/#register-a-module) and open a [pull request (PR)](/docs/infra-as-code-management/pipelines/operations/pr-automation), auto-generated testing pipelines are triggered at the project level. These pipelines, configured with the **use module registry** option, execute the `init`, `plan`, `apply`, and `destroy` commands to validate your module. Upon successful completion, your PR can proceed to merge.

Reviewer: This is only valid if they choose the integration testing, but a valid scenario would be to just use tofu test. Also, the customer can use their own pipelines.  I don't know how i would write it to be honest but i am not sure about this. We could remove this and let the next section to speak about it

Author: Fair enough, I'll remove it for now and can also add something more relevant later.
